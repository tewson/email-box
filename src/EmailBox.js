import juice from "juice";
import React from "react";
import { renderToStaticMarkup } from "react-dom/server";

export default ({ children, css }) => {
  const inlined = juice(
    `<style>${css}</style>${renderToStaticMarkup(children)}`
  );

  return (
    <div className="email-box" dangerouslySetInnerHTML={{ __html: inlined }} />
  );
};
