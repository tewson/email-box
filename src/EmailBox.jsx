import juice from 'juice';
import PropTypes from 'prop-types';
import React from 'react';
import { renderToStaticMarkup } from 'react-dom/server';

const EmailBox = ({ children, css }) => {
  const inlined = juice(
    `<style>${css}</style>${renderToStaticMarkup(children)}`,
  );

  return (
    <div className="email-box" dangerouslySetInnerHTML={{ __html: inlined }} />
  );
};

EmailBox.propTypes = {
  children: PropTypes.node.isRequired,
  css: PropTypes.string.isRequired,
};

export default EmailBox;
