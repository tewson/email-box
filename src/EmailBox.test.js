import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import EmailBox from "./EmailBox";

describe("EmailBox", () => {
  it("should be defined", () => {
    expect(EmailBox).toBeDefined();
  });

  it("should render correctly", () => {
    expect(toJson(shallow(<EmailBox />))).toMatchSnapshot();
  });
});
