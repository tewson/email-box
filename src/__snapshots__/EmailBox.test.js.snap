// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailBox should render children correctly 1`] = `
<div
  className="email-box"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<div><h1>Test</h1><p>Children</p></div>",
    }
  }
/>
`;

exports[`EmailBox should render correctly 1`] = `
<div
  className="email-box"
  dangerouslySetInnerHTML={
    Object {
      "__html": "",
    }
  }
/>
`;
